src/application/view_photo.c

```diff
@@ -0,0 +1,465 @@
+    BgConfig *bgConfig;
+    FieldViewPhoto *parent;
+    u8 filler_020[4];
+    MsgData *msgData;
+    MessageFormat *msgFormat;
+    String *strBuf;
+    String *exitMsg;
+    String *photoDescStringTemplates[2];
+    Window windows[2];
+    UnkStruct_ov01_021E7FDC spriteRender;
+    Sprite *sprites[3];
+    u8 animSpriteNo;
+    PhotoAlbumScroll scrollData;
+} ViewPhotoSysTaskData;
+
+static void SysTask_ViewPhoto(SysTask *task, void *taskData);
+static void ViewPhotoSysTask_Setup(ViewPhotoSysTaskData *viewPhoto);
+static void ViewPhotoSysTask_Teardown(ViewPhotoSysTaskData *viewPhoto);
+static ViewPhotoInputResponse ViewPhotoSysTask_HandleInput(ViewPhotoSysTaskData *viewPhoto);
+static ViewPhotoInputResponse ViewPhotoSysTask_GetKeyInput(ViewPhotoSysTaskData *viewPhoto);
+static ViewPhotoInputResponse ViewPhotoSysTask_GetTouchInput(ViewPhotoSysTaskData *viewPhoto);
+static void ViewPhotoSysTask_InitBgLayers(ViewPhotoSysTaskData *viewPhoto);
+static void ViewPhotoSysTask_ReleaseBgLayers(ViewPhotoSysTaskData *viewPhoto);
+static void ViewPhotoSysTask_LoadBgGraphics(ViewPhotoSysTaskData *viewPhoto);
+static void ViewPhotoSysTask_UnloadBgGraphics(ViewPhotoSysTaskData *viewPhoto);
+static void ViewPhotoSysTask_InitMessages(ViewPhotoSysTaskData *viewPhoto);
+static void ViewPhotoSysTask_ReleaseMessages(ViewPhotoSysTaskData *viewPhoto);
+static void ViewPhotoSysTask_InitWindows(ViewPhotoSysTaskData *viewPhoto);
+static void ViewPhotoSysTask_ReleaseWindows(ViewPhotoSysTaskData *viewPhoto);
+static void ViewPhotoSysTask_CreateSprites(ViewPhotoSysTaskData *viewPhoto);
+static void ViewPhotoSysTask_DeleteSprites(ViewPhotoSysTaskData *viewPhoto);
+static void ViewPhotoSysTask_AnimateButtonSelect(ViewPhotoSysTaskData *viewPhoto, int spriteNo);
+static BOOL ViewPhotoSysTask_IsButtonAnimPlaying(ViewPhotoSysTaskData *viewPhoto);
+static void formatPhotoFlavorText(PHOTO *a0, MessageFormat *msgFormat, String *strBuf, HeapID heapId, SaveData *saveData);
+static void ViewPhotoSysTask_DrawLyr3Icon(ViewPhotoSysTaskData *viewPhoto);
+static void ViewPhotoSysTask_PrintTextOnWindows(ViewPhotoSysTaskData *viewPhoto);
+static u8 Photo_CountValidMons(PHOTO *a0);
+
+static const WindowTemplate ov19_0225A04E[2] = {
+    {
+        .bgId = GF_BG_LYR_SUB_1,
+        .left = 24,
+        .top = 21,
+        .width = 8,
+        .height = 2,
+        .palette = 1,
+        .baseTile = 0x1F0
+    }, {
+        .bgId = GF_BG_LYR_SUB_1,
+        .left = 1,
+        .top = 8,
+        .width = 28,
+        .height = 8,
+        .palette = 10,
+        .baseTile = 0x110
+    }
+};
+
+static const TouchscreenHitbox ov19_0225A05E[] = {
+    { .rect = { 162, 188, 194, 254 } },
+    { .rect = { 24, 56, 80, 96 } },
+    { .rect = { 24, 56, 160, 176 }},
+    { { TOUCHSCREEN_RECTLIST_END } },
+};
+
+static const u16 ov19_0225A040[] = {
+    2, 3, 1, 0, -1, -1, NARC_resdat_resdat_00000072_bin
+};
+
+static const SpriteTemplate_ov01_021E81F0 ov19_0225A0C4[3] = {
+    {
+        1,
```

Comment:
maming these would be good, otherwise they're just out of context (I know you say use the vscode extension but I can't use that on github can I?)

Reply:
they're all unk fields at this point anyway

Reply:
understandable